src/app/containers/Preferences/Visualizer/Theme.js

```diff
@@ -57,63 +68,90 @@ const Theme = ({ state, actions }) => {
                             clearable={false}
                             menuContainerStyle={{ zIndex: 5 }}
-                            name="part"
-                            onChange={(part) => {
-                                setPart(part);
-                                actions.visualizer.handlePartChange();
-                            }}
-                            options={map(parts, (value) => ({
+                            name="theme"
+                            onChange={actions.visualizer.handleThemeChange}
+                            options={map(themes, (value) => ({
                                 value: value,
                                 label: value
                             }))}
                             searchable={false}
-                            value={{ label: part.value }}
+                            value={{ label: theme }}
                             valueRenderer={themeRenderer}
                             isDisabled={fileLoaded}
                         />
-                        <small>Choose which part to customize.</small>
+                        <small>Colours used when visualizing a G-Code file.</small>
                     </div>
-                )}
-            </Tooltip>
+                </Tooltip>
+            </Fieldset>
+            { theme === CUST_THEME && !fileLoaded &&
+                <Fieldset legend="Colours">
+                    <div className={styles.addMargin}>
+                        <Tooltip content="Save your changes" location="default">
+                            <button
+                                className={styles.saveColour}
+                                type="button"
+                                onClick={() => {
+                                    actions.visualizer.handleCustThemeChange(themeColours);
+                                }}
+                            >
+                            Save
+                            </button>
+                        </Tooltip>
+                        <Tooltip content="Click on the colour circles to change the colour for that component" location="default">
+                            {
+                                parts.map((value, i) => {
+                                    return (
+                                        <div key={i} className={styles.colorContainer}>
+                                            <span className={styles.first}>{value === G1_PART ? 'G1-G3' : value}</span>
+                                            <div className={styles.dots}></div>
+                                            <div className={styles.colorDisplay}>
+                                                <span>{themeColours.get(value)}</span>
+                                                <ColorCircle part={value} onClick={openModal} colour={themeColours.get(value)} index={i}/>
+                                            </div>
+                                        </div>
```

Comment:
Maybe making the whole section clickable would be easier for the user since I have a feeling some people will complain that they are having a hard time selecting the color circle